ClinicManagement/src/ClinicManagement.Core/Interfaces/IMessagePublisher.cs

```diff
@@ -1,9 +1,10 @@
-﻿using PluralsightDdd.SharedKernel;
+﻿using System.Threading.Tasks;
+using PluralsightDdd.SharedKernel;
 
 namespace ClinicManagement.Core.Interfaces
 {
   public interface IMessagePublisher
   {
-    void Publish(BaseIntegrationEvent applicationEvent);
+    Task Publish(object applicationEvent);
```

Comment:
We don't have any restriction on object type we can use here?

Reply:
I wasn't thinking we needed a restriction, but let me know what you have in mind.

Because of how MassTransit handles [inheritance](https://masstransit.io/documentation/concepts/messages#message-inheritance) it actually creates RabbitMQ exchanges for each derived type and will those derived messages to those exchanges (1 for the concreate class, then once for `BaseIntegrationEvent`, then again for `INotification`).

I opted for simpler.
But I'm not opposed to reverting back to `BaseIntegrationEvent`.

At that point, would you want me to remove `INotification` from the inheritance chain, since we're no longer publishing these with MediatR.

Reply:
I wasn't aware of MT's behavior regarding exchanges. Are you saying that a deeply inherited type would be published to an exchange for every parent it has?
Object -> BaseIntegrationEvent -> ConcreteEvent would involve 3 exchanges in RabbitMQ not just `ConcreteEvent` ?

If we're not using MediatR for certain messages/contracts we definitely can remove `INotification` from them.

Reply:
That's correct:
- ConcreteEvent
- BaseIntegrationEvent
- INotification

So would you like me to inherit from BaseIntegartion event for all the message contracts? (knowing that it will create one extra exhange?)